### DISCUSSION TITLE
Is it possible to limit ripgrep results by also filtering the file name

### DISCUSSION THREAD

**Discussion by Hubro** [nvim-telescope/telescope.nvim#1367] [Q&A]
Most often when I run `live_grep`, I'm not really interested in results from *all files*. Usually I only want results from a certain file type, or from files in a certain subdirectory.

Is there anything I can write in the `builtin.live_grep` search box that enables me to also filter for filenames?

I know there is precedence for this, because when using `builtin.lsp_document_symbols`, I can write (for example) `:class:` to only display class symbols.

It would be awesome if I could write: (for example) `:.xml:foo` to search for the text string "foo" in files that contain ".xml" in the filename.

Is there any such mechanism currently?

**Comment by bastilimbach** [comment]
I'm also looking for a solution. A workaround would be to add/remove the folder to the `.rgignore` file prior to the search. However, this is quite cumbersome.
